### PR TITLE
jNery/issue1502_Legend_Windows_DesignUpdate

### PR DESCRIPTION
**Pull request by jNery**
- Added Legend (new design), moved LayerLegend (existing toolkit control), refactored TableOfContents code into LayerList (base for Legend). 
- Updated LegendSample to use new Legend, added LayerLegend sample (existing toolkit control).
- Separated SymbolDisplay, LayerLegend, LayerList, Legend implementations for Windows and Xamarin. 
- Added placeholders for iOS/Android SymbolDisplay, LayerLegend, LayerList, and Legend controls.

### CODE REVIEW

File: src/Esri.ArcGISRuntime.Toolkit/UWP/UI/Controls/Legend/Legend.Theme.xaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                </DataTemplate>
+            </Setter.Value>
+        </Setter>
+        <Setter Property="Template">
+            <Setter.Value>
+                <ControlTemplate TargetType="controls:Legend">
+                    <Border BorderBrush="{TemplateBinding BorderBrush}"
+                            BorderThickness="{TemplateBinding BorderThickness}"
+                            Background="{TemplateBinding Background}">
+                        <ScrollViewer>
```
Comment (dotMorten) [inline src/Esri.ArcGISRuntime.Toolkit/UWP/UI/Controls/Legend/Legend.Theme.xaml]: No scrollviewer

File: src/Esri.ArcGISRuntime.Toolkit/UWP/UI/Controls/Legend/Legend.Theme.xaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                          ItemTemplate="{StaticResource LeafItemTemplate}"/>
+        </StackPanel>
+    </DataTemplate>
+    <DataTemplate x:Key="TrunkItemTemplate">
+        <StackPanel Visibility="{Binding DisplayLegend, Converter={StaticResource VisibilityConverter}}">
+            <TextBlock Text="{Binding LayerContent.Name}"
+                       Margin="10,0" 
+                       Visibility="{Binding IsSublayer, Converter={StaticResource VisibilityConverter}}"/>
+            <ItemsControl ItemsSource="{Binding Sublayers}" ItemTemplate="{StaticResource BranchItemTemplate}"/>
+        </StackPanel>
```
Comment (dotMorten) [inline src/Esri.ArcGISRuntime.Toolkit/UWP/UI/Controls/Legend/Legend.Theme.xaml]: This is limited to 3 levels. We need an approach for N levels.

File: src/Esri.ArcGISRuntime.Toolkit/UWP/UI/Controls/SymbolDisplay/SymbolDisplay.Theme.xaml [hunk 3]
```diff
@@ -7,7 +7,7 @@
         <Setter Property="Template">
             <Setter.Value>
                 <ControlTemplate TargetType="controls:SymbolDisplay">
-                    <Image x:Name="image" />
+                    <Image x:Name="image" Stretch="Fill" MaxWidth="40" MaxHeight="40"/>
```
Comment (dotMorten) [inline src/Esri.ArcGISRuntime.Toolkit/UWP/UI/Controls/SymbolDisplay/SymbolDisplay.Theme.xaml]: MaxWidth/Height should be on the SymbolDisplay control, not it's internal items.
Ie
```
<Setter Property="MaxHeight" Value="40" />
```
However I don't think we should make that default. We should display the symbol the user gave us, and they are free to set `<esri:SymbolDisplay MaxHeight="40" />`

File: src/Esri.ArcGISRuntime.Toolkit/Shared/UI/Controls/SymbolDisplay/SymbolDisplay.Windows.cs [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            }
+            else
+            {
+#pragma warning disable ESRI1800 // Add ConfigureAwait(false) - This is UI Dependent code and must return to UI Thread
+                try
+                {
+                    var scale = GetScaleFactor();
+                    var imageData = await Symbol.CreateSwatchAsync(scale * 96);
+                    img.Width = imageData.Width / scale;
+                    img.Height = imageData.Height / scale;
```
Comment (dotMorten) [inline src/Esri.ArcGISRuntime.Toolkit/Shared/UI/Controls/SymbolDisplay/SymbolDisplay.Windows.cs]: Use MaxWidth/Height

**Comment by jNery** [comment]
Note to self - from review feedback 
1 issue left to solve: Nested ListView is limited to 3-levels.

File: src/Esri.ArcGISRuntime.Toolkit/Shared/UI/Controls/LayerLegend/LayerLegend.Windows.cs [hunk 5]
```diff
@@ -69,6 +69,7 @@ private void Refresh()
                 if ((LayerContent as ILoadable).LoadStatus != LoadStatus.Loaded)
                 {
                     (LayerContent as ILoadable).Loaded += Layer_Loaded;
+                    (LayerContent as ILoadable).LoadAsync();
```
Comment (dotMorten) [inline src/Esri.ArcGISRuntime.Toolkit/Shared/UI/Controls/LayerLegend/LayerLegend.Windows.cs]: Isn't this generating a warning?

File: src/Esri.ArcGISRuntime.Toolkit/Shared/UI/Controls/Legend/ItemTemplateSelector.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+#else
+using System.Windows;
+using System.Windows.Controls;
+#endif
+
+namespace Esri.ArcGISRuntime.Toolkit.UI.Controls
+{
+    public class ItemTemplateSelector : DataTemplateSelector
```
Comment (dotMorten) [inline src/Esri.ArcGISRuntime.Toolkit/Shared/UI/Controls/Legend/ItemTemplateSelector.cs]: No xml doc. Should this be hidden from intellisense? Does it need to be public?
Also we should pick a more specific name for this.

**Pull request merged**